File: tests/test_forms.py
from playwright.sync_api import sync_playwright

def test():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        try:
            print("Navigating to login...")
            page.goto('http://localhost:4321/login')
            page.wait_for_load_state('networkidle')
            print("Login page loaded.")
            page.screenshot(path='login.png', full_page=True)
            
            # Intento de login usando credenciales genéricas para ver si es posible
            print("Intento login...")
            # Llenamos los inputs refactorizados de login
            page.fill('input[name="username"]', 'admin')
            page.fill('input[name="password"]', 'admin')
            page.click('button[type="submit"]')
            page.wait_for_load_state('networkidle')
            print("Después del login.")
            page.screenshot(path='post_login.png', full_page=True)
            
        except Exception as e:
            print("Error:", e)
        finally:
            browser.close()

if __name__ == '__main__':
    test()

File: tests/test_forms.mjs
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  try {
    console.log("Navegando al login...");
    await page.goto('http://localhost:4321/login');
    await page.waitForLoadState('networkidle');
    console.log("Login cargado.");
    await page.screenshot({ path: 'login.png', fullPage: true });

    console.log("Intentando login...");
    await page.fill('input[name="username"]', 'admin');
    await page.fill('input[name="password"]', 'admin');
    await page.click('button[type="submit"]');
    await page.waitForLoadState('networkidle');
    console.log("Después de intentar el login.");
    await page.screenshot({ path: 'post_login.png', fullPage: true });
  } catch (error) {
    console.error("Error:", error);
  } finally {
    await browser.close();
  }
})();
